CMake 2.8.11 fails for tests

Since #1190 we create test output subdirectories and for that use the CMAKE command: `GET_FILENAME_COMPONENT(_dir ${_output} DIRECTORY)`
unfortunately this was changed between cmake 2.8.11 and 2.8.12, and versions <= 2.8.11 only accept:
`GET_FILENAME_COMPONENT(_dir ${_output} PATH)`
Now 2.8.11 is quite old (I use it, because it is preinstalled on a cluster), and I can easily replace it with an own build version, but we make no requirements on cmake for ASPECT itself, and the deal.II website states anything newer than 2.8.8 should work, so I was just wondering if we should do one or more of the following:
- mention this requirement? every desktop system from Ubuntu 14.04 on already has cmake 2.8.12 packaged, so it will only affect users from very conservative distributions and cluster systems with prepackaged old cmakes
- provide a better error message? Something like: require newer cmake version (current message is:  
`CMake Error at tests/CMakeLists.txt:182 (GET_FILENAME_COMPONENT):
   get_filename_component unknown component DIRECTORY)`
- Add a workaround to the CMakeLists.txt for older versions?

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -179,7 +179,15 @@ FOREACH(_test ${_tests})
     FILE(GLOB_RECURSE _outputs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}/${_test}/
 	${CMAKE_CURRENT_SOURCE_DIR}/${_test}/[a-zA-Z0-9]*)
     FOREACH(_output ${_outputs})
-      GET_FILENAME_COMPONENT(_dir ${_output} DIRECTORY)
+
+      # get the directory name we should work on; for cmake <2.8.11, the
+      # subcommand was PATH (legacy), for everything after we can use
+      # DIRECTORY
+      IF (${CMAKE_VERSION} VERSION_LESS "2.8.12")
+        GET_FILENAME_COMPONENT(_dir ${_output} PATH)
+      ELSE()
+        GET_FILENAME_COMPONENT(_dir ${_output} DIRECTORY)
+      ENDIF()
       IF(NOT  ${_dir} STREQUAL "" )
         FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_dir})
       ENDIF()

